ssl/t1_lib.c

```diff
@@ -2743,6 +2728,30 @@ int ssl_check_clienthello_tlsext_late(SSL *s)
     } else
         s->tlsext_status_expected = 0;
 
+    if (s->ctx->alpn_select_cb != NULL) {
+        const unsigned char *selected = NULL;
+        unsigned char selected_len = 0; /* yes, this is unsigned char */
```

Comment:
Yeah, the callback API takes an unsigned char as a length, which seems very strange. This was also copied from tls1_alpn_handle_client_hello().